jetbrains-core/src/software/aws/toolkits/jetbrains/ui/clouddebug/ArtifactMappingPopup.kt

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.clouddebug
+
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.ui.popup.ListPopup
+import com.intellij.openapi.ui.popup.PopupStep
+import com.intellij.openapi.ui.popup.util.BaseListPopupStep
+import com.intellij.ui.CellRendererPanel
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.popup.list.ListPopupImpl
+import com.intellij.util.ui.JBUI
+import net.miginfocom.swing.MigLayout
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.trace
+import software.aws.toolkits.jetbrains.services.ecs.execution.ArtifactMapping
+import software.aws.toolkits.resources.message
+import java.awt.Component
+import javax.swing.Icon
+import javax.swing.JList
+import javax.swing.JPanel
+import javax.swing.ListCellRenderer
+
+/**
+ * Popup represents values from Artifact Mapping table
+ */
+class ArtifactMappingPopup {
+
+    companion object {
+        private const val MAX_ROW_COUNT = 5
+
+        fun createPopup(
+            artifactMappingItems: List<ArtifactMapping>,
+            onSelected: (ArtifactMapping?) -> Unit
+        ): ListPopup {
+            val step = ArtifactMappingPopupStep(artifactMappingItems, onSelected)
+            val popup = JBPopupFactory.getInstance().createListPopup(step, MAX_ROW_COUNT)
+
+            val popupToUpdate = popup as? ListPopupImpl
+                ?: throw IllegalStateException("Unable to cast popup in type: '${popup.javaClass}' to ${ListPopupImpl::class.qualifiedName}")
+
+            popupToUpdate.list.setCellRenderer(PathMappingPopupCellRenderer())
+            return popupToUpdate
```

Comment:
Does this work to avoid the hard cast to Impl
```suggestion
            return JBPopupFactory.getInstance().createPopupChooserBuilder(artifactMappingItems)
            .setTitle(message("cloud_debug.run_configuration.generate_automatically.popup_title")
            .setRenderer(PathMappingPopupCellRenderer())
            .setVisibleRowCount(MAX_ROW_COUNT)
            .setItemSelectedCallback(onSelected)
            .createPopup()
```